registry/registry.go

```diff
@@ -0,0 +1,166 @@
+package registry
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/http"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/remote/transport"
+)
+
+// Platform represents an OS/architecture combination
+type Platform struct {
+	OS           string
+	Architecture string
+}
+
+func (p Platform) String() string {
+	return fmt.Sprintf("%s/%s", p.OS, p.Architecture)
+}
+
+// Image contains information about an image in a specific registry
+type Image struct {
+	Exists    bool
+	Platforms map[Platform]bool
+}
+
+// ImageInfo contains information about an image across registries
+type ImageInfo struct {
+	Reference  name.Reference
+	OSSImage   Image
+	PrimeImage Image
+}
+
+// Client provides methods for interacting with a container registry
+type Client interface {
```

Comment:
Are there multiple implementations of this interface?

Reply:
No reason for this interface, removed